core/src/wsv.rs

```diff
@@ -1140,10 +1140,10 @@ impl WorldStateView {
     /// then *trigger* will be executed on the **current** block
     /// - If this method is called by ISI inside *trigger*,
     /// then *trigger* will be executed on the **next** block
-    pub fn execute_trigger(&self, trigger_id: TriggerId, authority: AccountId) {
+    pub fn execute_trigger(&self, trigger_id: TriggerId, authority: &AccountId) {
         let event = ExecuteTriggerEvent {
             trigger_id,
-            authority,
+            authority: authority.clone(),
```

Comment:
It's not always preferable to push `.clone` up the call stack. 

Here we were cloning `AccountId` every single time before executing any instruction. However, cloning is only necessary for one specific instruction `Register<AssetDefinition>` which is a fairly rare instruction

Reply:
How `Register<AssetDefinition>` is related to `execute_trigger()`?

Reply:
ah, ok. In this case I think it makes a more consistent API

Reply:
> It's unclear from the outside why would you want to take ownership of authority

Well, okay, I'm not always like that idea because it eliminates the place for optimizations on caller side sometimes, but let's keep it your way